src/type.cpp

```diff
@@ -158,6 +160,35 @@ std::string StructType::id() const noexcept {
   return id_;
 }
 
+std::size_t StructType::offset(const std::string& id) const {
+  std::size_t offset = 0;
+  for (auto i = std::size_t{0}, e = fields_.size(); i < e; ++i) {
+    const auto& field = fields_.at(i);
+    if (field->id == id) {
+      return offset;
+    }
+
+    offset += field->type->size();
+  }
+
+  throw std::runtime_error("member not found in struct!");
+}
+
+std::size_t StructType::offset(std::size_t index) const {
+  if (index >= fields_.size()) {
+    throw std::runtime_error("index out of bound!");
+  }
+
+  return std::accumulate(
+      fields_.cbegin(), std::next(fields_.cbegin(), (long)index),
```

Comment:
I find the _end_ too heavy, let's extract it:
```suggestion
  auto end = std::next(fields_.cbegin(), (long)index);
  return std::accumulate(
      fields_.cbegin(), end,
```